index.js

```diff
@@ -370,7 +376,13 @@ class ServerlessAmplifyPlugin {
             }
         }
 
-        const identityPool = resources.find(r => r.ResourceType === 'AWS::Cognito::IdentityPool');
+        const identityPools = resources.filter(r => r.ResourceType === 'AWS::Cognito::IdentityPool');
+        const identityPoolForUserPool = fileDetails.hasOwnProperty('appClient')
+            ? identityPools.find(r => r.metadata.CognitoIdentityProviders &&
+                r.metadata.CognitoIdentityProviders.some(({ClientId}) => ClientId === config.aws_user_pools_web_client_id)
+            )
+            : undefined;
+        const identityPool = identityPoolForUserPool || identityPools[0];
```

Comment:
Ditto for this.  In the tests, not having an identity pool threw an error, so I definitely want to clean this up before including.

Reply:
I tested on a stack without ID pools, it gave the intended behavior. I also tried removing the `appClient` parameter, and it still worked as expected.

Can I find the `serverless.yml` file for the failing test somewhere? I'd love to get to the bottom of this.